frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -149,12 +165,38 @@ export const useValueSetsTable = <T extends ValueSet | ValueSetRow>(
         }
     }
 
+    const memoizedDataFetch = useCallback(
+        () =>
+            dataFetch(
+                toFetchParams(getTableDataConfig, {
+                    segments: {
+                        tableName: dataTableName,
+                        version: `${versionData!!.version!}`, // number to string,
+                    },
+                })
+            ),
+        [dataFetch, versionData, dataTableName]
+    );
+
     // not entirely accurate typing. What is sent back by the api is actually ApiValueSet[] rather than ValueSet[]
     // does not seem entirely worth it to add the complexity needed to account for that on the frontend, better
     // to make the API conform better to the frontend's expectations. TODO: look at this when refactoring the API
     const { error: valueSetError, data: valueSetData } = useQuery<Array<T>>(
         ["lookupTable", versionData?.version, dataTableName],
-        () => getLatestData<Array<T>>(dataTableName, versionData!.version!!),
+        // note that fetch function is an inlined anonymous function in order to allow for
+        // useQuery to pick up changes to variables that the function is dependent on
+        // if we pass in a predefined function, that exact function will be called on every render
+        // and no dynamic variables will be updated
+        memoizedDataFetch,
+        // () =>
+        //     dataFetch(
```

Comment:
it looks like the memoized version worked? Just some rogue comments leftover

Reply:
yep, good catch. The useCallback makes the usage much easier to understand, so I can get rid of that long comment too